Nuevo tipo de dato String

ronald1512 opened the pull request:
Esta es mi propuesta de mejora: Se agregó el nuevo tipo de dato String, con ello ahora se pueden hacer declaraciones de dicho tipo, así mismo asignaciones. También se tomaron en cuenta todas las validaciones correspondientes para que funcione de manera correcta. Lamento los últimos 2 pull request que cerré, lo hice porque no había seguido correctamente las instrucciones, por favor si cometí algún error, siéntete libre de hacerme una corrección, muchas gracias. Soy Ronald Romero 201701048 OLC2 sección B+

ronald1512 committed:
Update entrada.txt

En aqui actualicé el archivo de entrada para validar el funcionamiento del Nuevo tipo de dato String en declaración y asignacion.

ronald1512 committed:
Update gramatica.jison and gramatica.js

Aqui hice los cambios necesarios en la gramatica para poder declarar y asignar con tipo de dato String. Ejecuté el archivo compilar.sh con start compilar.sh y generó el archivo gramatica.js

ronald1512 committed:
Update instrucciones.js

Aqui solamente modifiqué que nuevoDeclaracion acepta el tipo de declaracion, que puede ser string o number. De esta forma al ejecutar la instruccion ya se sabe de que tipo de dato se trata.

ronald1512 committed:
Update interprete.js

En este archivo se van ejecutando las instrucciones capturadas, así que aqui agregue las validaciones de tipos en la asignacion y en la declaracion ahora se envía el tipo de dato, anteriormente no se recibía este parámetro.

ronald1512 committed:
Update tabla_simbolos.js

Se agregó como TIPO_DATO string y se realizaron algunas validaciones para que funcionen correctamente las funciones actualizar y obtener.

rjanixz reviewed (changes requested):
@ronald1512 hemos comenzado a hacer merge de los pull request aprobados, lamentablemente con los cambios que ya fuero agregados al repositorio, tus cambios entran en conflictos, podrías por favor corregirlos?

Si tienes alguna duda no dudes en hacerme saber.

Saludos

ronald1512 committed:
Merge branch 'master' of https://github.com/ericknavarro/InterpreteSencilloJisonJS

ronald1512 commented:
Buenas Rainman, ya resolví los confilctos. Me cuentas cualquier cosa

ronald1512 committed:
Prueba de archivo de entrada con todos los cambios

Modifiqué la parte del i++ del for hecho por algun compañero, ya que no tomaba encuenta el envío del tipo del valor a asignar en la tabla de simbolos, lo que ocasionaba que el programa lanzara un error. Con estos cambios he comprobado que todo funciona correctamente.

rjanixz commented:
@ronald1512 perdona los inconvenientes, aún hay conflictos con el archivo gramatica.js.

Podes resolverlos por favor?
